examples/lighting-app/silabs/src/ColorFormat.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ColorFormat.h"
+
+#include <math.h>
+
+// define a clamp macro to substitute the std::clamp macro which is available from C++17 onwards
+#define clamp(a, min, max) ((a) < (min) ? (min) : ((a) > (max) ? (max) : (a)))
+
+RgbColor_t HsvToRgb(HsvColor_t hsv)
+{
+    RgbColor_t rgb;
+
+    uint8_t region, p, q, t;
+    uint32_t h, s, v, remainder;
+
+    if (hsv.s == 0)
+    {
+        rgb.r = rgb.g = rgb.b = hsv.v;
+    }
+    else
+    {
+        h = hsv.h;
+        s = hsv.s;
+        v = hsv.v;
+
+        region    = h / 43;
+        remainder = (h - (region * 43)) * 6;
+        p         = (v * (255 - s)) >> 8;
+        q         = (v * (255 - ((s * remainder) >> 8))) >> 8;
+        t         = (v * (255 - ((s * (255 - remainder)) >> 8))) >> 8;
+        switch (region)
+        {
+        case 0:
+            rgb.r = v, rgb.g = t, rgb.b = p;
+            break;
+        case 1:
+            rgb.r = q, rgb.g = v, rgb.b = p;
+            break;
+        case 2:
+            rgb.r = p, rgb.g = v, rgb.b = t;
+            break;
+        case 3:
+            rgb.r = p, rgb.g = q, rgb.b = v;
+            break;
+        case 4:
+            rgb.r = t, rgb.g = p, rgb.b = v;
+            break;
+        case 5:
+        default:
+            rgb.r = v, rgb.g = p, rgb.b = q;
+            break;
+        }
+    }
+
+    return rgb;
+}
+
+RgbColor_t XYToRgb(uint8_t Level, uint16_t currentX, uint16_t currentY)
+{
+    // convert xyY color space to RGB
+
+    // https://www.easyrgb.com/en/math.php
+    // https://en.wikipedia.org/wiki/SRGB
+    // refer https://en.wikipedia.org/wiki/CIE_1931_color_space#CIE_xy_chromaticity_diagram_and_the_CIE_xyY_color_space
+
+    // The currentX/currentY attribute contains the current value of the normalized chromaticity value of x/y.
+    // The value of x/y shall be related to the currentX/currentY attribute by the relationship
+    // x = currentX/65536
+    // y = currentY/65536
+    // z = 1-x-y
+
+    RgbColor_t rgb;
+
+    float x, y, z;
+    float X, Y, Z;
+    float r, g, b;
+
+    x = (static_cast<float>(currentX)) / 65535.0f;
+    y = (static_cast<float>(currentY)) / 65535.0f;
```

Comment:
can all the constant values here be declared as constexpr with a relevant name

Reply:
Added constexpr for constant values and replaced with it.